x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/use_columns.tsx

```diff
@@ -51,6 +51,11 @@ interface ColumnsProps {
   hasPermissions: boolean;
 }
 
+const durationFormat = new DurationFormat({
+  outputFormat: 'humanizePrecise',
+  outputPrecision: 0,
+});
+
```

Comment:
As per plugin usage examples, we're supposed to use the `fieldFormats` plugin instance to initialize formatters:
 https://github.com/elastic/kibana/blob/1c8e0ed785146f064b14b936f2498c5a15a4b1a7/examples/field_formats_example/public/examples/1_using_existing_format.ts#L23-L25

I think this is important as formatters should respect user-defined settings by default and overwrite only what is needed for a specific use case. With the current implementation, user preferences don't get passed by default to the formatters. Moreover, having different formatters instantiated in various places across our codebase could become hard to maintain. I would suggest looking into how that is implemented in other areas, like the ML plugin, for example, where formatters and their options are abstracted:

https://github.com/elastic/kibana/blob/87fa95d49e5cdd80e8ebffbb078009b0f4636480/x-pack/plugins/ml/public/application/trained_models/models_management/expanded_row.tsx#L52-L53

Reply:
I'm aligned with @banderror here. It will be awkward to have only one column regarding user settings. Though it makes sense to discuss in general.